TsRandomizer/Screens/GameplayScreen.cs

```diff
@@ -229,6 +237,7 @@ bool IsRoomChanged()
 			currentRoom = LevelReflected.CurrentRoom;
 
 			ExceptionLogger.SetLevelContext(Level.ID, currentRoom.ID);
+			ExtraEarringsXP = Save.GetSettings().ExtraEarringsXP.Value;
```

Comment:
This method is to see if the room was changed, it should not be used to set a value like this